.github/workflows/build.yml

```diff
@@ -0,0 +1,48 @@
+name: Build and Publish
+
+on:
+    workflow_dispatch:
+    pull_request:
+    push:
+        branches:
+            - master
```

Comment:
we can also add `- main` to be future proof

Reply:
Great suggestion, I'll add it